File: server/router/cloudpods/network/eip.go
package network

import (
	v1 "ygpt/server/api/v1"

	"github.com/gin-gonic/gin"
)

type NetEipRouter struct {
}

func (s *NetEipRouter) InitNetworkEipRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/network/eip")
	ipApi := v1.ApiGroupApp.CloudApiGroup.NetApiGroup.NetworkEipApi

	{
		dataRouter.POST("getNetEipList", ipApi.GetEipList)            // 获取eip列表
		dataRouter.POST("addNetEip", ipApi.AddEip)                    // 新增eip
		dataRouter.POST("deleteNetEip", ipApi.DeleteEip)              // 删除eip
		dataRouter.POST("deleteNetEipByIds", ipApi.DeleteEipByIds)    // 删除eip
		dataRouter.POST("bindEipServer", ipApi.BindEipServer)         // 绑定
		dataRouter.POST("unbindEipServer", ipApi.UnbindEipServer)     // 解绑
		dataRouter.POST("getBindServerList", ipApi.GetBindServerList) // 获取可绑定云主机
		dataRouter.POST("modEipBandWidth", ipApi.ModEipBandWidth)     // 修改带宽
	}
}

File: web/src/api/cloudpods/network/eip.js
import request from '@/utils/request'
//获取
export function getNetEipList(params){
    return request({
        url:'cloud/network/eip/getNetEipList',
        method:'post',
        data: params
    })
}
//新增
export function addNetEip(params){
    return request({
        url:'cloud/network/eip/addNetEip',
        method:'post',
        data: params
    })
}
//删除
export function deleteNetEip(params){
    return request({
        url:'cloud/network/eip/deleteNetEip',
        method:'post',
        data: params
    })
}
//批量删除
export function deleteNetEipByIds(params){
    return request({
        url:'cloud/network/eip/deleteNetEipByIds',
        method:'post',
        data: params
    })
}
//绑定
export function bindEipServer(params){
    return request({
        url:'cloud/network/eip/bindEipServer',
        method:'post',
        data: params
    })
}
//解绑
export function unbindEipServer(params){
    return request({
        url:'cloud/network/eip/unbindEipServer',
        method:'post',
        data: params
    })
}
//获取可绑定云主机
export function getBindServerList(params){
    return request({
        url:'cloud/network/eip/getBindServerList',
        method:'post',
        data: params
    })
}
//修改带宽
export function modEipBandWidth(params){
    return request({
        url:'cloud/network/eip/modEipBandWidth',
        method:'post',
        data: params
    })
}